oshi-core/src/main/java/oshi/util/ConfigUtil.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * OSHI (https://github.com/oshi/oshi)
+ *
+ * Copyright (c) 2010 - 2019 The OSHI Project Team:
+ * https://github.com/oshi/oshi/graphs/contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package oshi.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The global configuration utility.
+ * 
+ * @author cilki
+ * @since 4.0.0
+ */
+public final class ConfigUtil {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ConfigUtil.class);
+
+    private static final Properties configuration = new Properties();
+
+    static {
+        try (InputStream in = ConfigUtil.class.getResourceAsStream("/default.properties")) {
+            if (in != null)
+                configuration.load(in);
+            else
+                LOG.warn("Default configuration not found");
+        } catch (IOException e) {
+            LOG.warn("Failed to load default configuration");
+        }
+    }
+
+    /**
+     * Get the {@code String} property associated with the given key.
+     * 
+     * @param key The property key
+     * @return The property value or {@code null} if not found
+     */
+    public static String get(String key) {
+        return configuration.getProperty(key);
```

Comment:
This seems to be the only place to check whether a property even exists (`null` return if not).  I wonder if we really need the other convenience getter methods (particularly the ones without user-supplied defaults), since they  essentially do the same thing a user could do (but without giving the user any indication of whether the value was actually the default).

Java 8+ adds the ability for a default property list so we could actually guarantee never to get null....